lib/oauth2/client.ex

```diff
@@ -358,6 +361,21 @@ defmodule OAuth2.Client do
     end
   end
 
+  @doc """
+  Add authentication context by auth_scheme param
+
+  The auth_scheme parameter is compatible with "auth_header" and "request_body", but also refers to the OIDC specification.
+  ref. https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication
+  """
+  @spec auth_scheme(t, binary) :: t
+  def auth_scheme(client, scheme) when scheme in ["auth_header", "client_secret_basic"],
```

Comment:
```suggestion
  def auth_method(client, scheme) when scheme in ["auth_header", "client_secret_basic"],
```

From the spec:

> This section defines a set of Client Authentication methods

They call them `Auth Methods` so maybe we should use that wording? Not sure

Reply:
I also think auth_scheme is more appropriate.
I will take this in and add the relevant diffs.

Reply:
@ritou my apologize for the mess I created,

It seems that such changes will cause breaking changes, no? If so, it will be much better to keep it as `auth_scheme` as it was before; definitely, we should avoid any breaking change.

I am sorry.